tests/tel2puml/find_unique_graphs/test_json_data_converter.py

```diff
@@ -0,0 +1,226 @@
+        assert len(header_dict) == 1
+        assert header_dict[path1] == flatdict.FlatterDict(value1)
+
+        # Test updating with a dict complex path
+        header_dict = {}
+        path2 = "key1::key2"
+        value2 = {"key3": "value"}
+        _update_header_dict(header_dict, path2, value2)
+        assert len(header_dict) == 1
+        assert header_dict["key1"] == flatdict.FlatterDict(value2)
+
+        # Test updating with a list simple path
+        header_dict = {}
+        path3 = "key1:key2"
+        value3 = [{"key3": "value"}]
+        _update_header_dict(header_dict, path3, value3)
+        assert len(header_dict) == 1
+        assert header_dict[path3] == flatdict.FlatterDict(value3)
+
+        # Test updating with a list complex path
+        header_dict = {}
+        path4 = "key1::key2"
+        value4 = [{"key3": "value"}]
+        _update_header_dict(header_dict, path4, value4)
+        assert len(header_dict) == 1
+        assert header_dict["key1"] == flatdict.FlatterDict(value4)
+
+        # Test updating with a string simple path
+        header_dict = {}
+        path5 = "key1:key2"
+        value5 = "value"
+        _update_header_dict(header_dict, path5, value5)
+        assert len(header_dict) == 1
+        assert header_dict[path5] == "value"
+
+        # Test updating with a string complex path
+        header_dict = {}
+        path6 = "key1::key2"
+        value6 = "value"
+        _update_header_dict(header_dict, path6, value6)
+        assert len(header_dict) == 1
+        assert header_dict["key1"] == "value"
+
+    @staticmethod
+    def test_build_nested_dict() -> None:
+        """Tests the function _build_nested_dict"""
+
+        # Test multiple segments
+        path_segments1 = ["key1", "key2", "key3"]
+        parsed_data1 = {"key4": "value4"}
+        expected1 = {"key2": {"key3": {"key4": "value4"}}}
+        assert _build_nested_dict(path_segments1, parsed_data1) == expected1
+
+        # Test with string data
+        path_segments2 = ["key1", "key2", "key3"]
+        parsed_data2 = "string_value"
+        expected2 = {"key2": {"key3": "string_value"}}
+        assert _build_nested_dict(path_segments2, parsed_data2) == expected2
+
+        # Test with list of dicts3
+        path_segments3 = ["key1", "key2", "key3"]
+        parsed_data3 = [{"key4": "value4"}, {"key5": "value5"}]
+        expected3 = {
+            "key2": {"key3": [{"key4": "value4"}, {"key5": "value5"}]}
+        }
+        assert _build_nested_dict(path_segments3, parsed_data3) == expected3
+
+        # Test error handling with one path segment. In reality there will
+        # always be more than one path segment since the path requires '::'
+        # for the function to be reached
+        path_segments4 = ["key1"]
+        parsed_data4 = 123
+        with pytest.raises(TypeError):
+            _build_nested_dict(
+                path_segments4, parsed_data4  # type: ignore[arg-type]
```

Comment:
Can just make the above astring so we don't have to ignore the mypy error

Reply:
Updated to string